### PR TITLE
Swap node sass with dart sass

### PR DESCRIPTION
**Pull request by notlee**
- Swap lib sass for dart sass (continue to test node sass)
- Remove node v8 support
- Fix demo sass source maps

I dropped node v8 support as it is no longer maintained and throwing errors related to the new dart-sass library.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -131,7 +131,7 @@ obt init o-my-new-component
 ### Migrating from 7.X.X to 8.X.X
 
 
-OBT no longer supports NodeJS v6 because it uses async functions. To use this version of OBT, you will need NodeJS v8 or above.
+OBT no longer supports NodeJS v6 because it uses async functions. To use this version of OBT, you will need NodeJS v10 or above.
```
Comment (JakeChampion) [inline README.md]: This should be under 8.xx to 9.xx as we are already on version 8

Comment (notlee) [reply README.md]: this was a mistake I need to write a new migration guide, thanks :)

**Comment by chee** [comment]
very exciting!

File: test-build/main.css [hunk 2]
```diff
@@ -0,0 +1,4 @@
+div {
```
Comment (JakeChampion) [inline test-build/main.css]: I don't see this file being used anywhere, should it be ignored from the repo?

File: lib/tasks/build-sass.js [hunk 3]
```diff
@@ -8,81 +8,118 @@ const outputFile = denodeify(fs.outputFile);
...
+								`\n${path.join(cwd, file)}:${line}:${column}\n`;
+						}
+						// Forward Sass error.
+						throw new Error(errorMessage);
+					}
+
+					return result.stdout;
+				}).then(css => {
+					// post css does not parse the charset unless it is also base64.
+					// Remove the charset as a workaround, and remove this code
```
Comment (JakeChampion) [inline lib/tasks/build-sass.js]:
```suggestion
					// TODO: Remove the charset as a workaround, and remove this code
```

File: test-build/main.css (hunk 2 above)
Comment (notlee) [reply test-build/main.css]: It shouldn't be there at all. I'm not sure what created it. Going to assume it was my mistake and remove

File: lib/tasks/build-sass.js [hunk 4]
```diff
@@ -8,81 +8,118 @@ const outputFile = denodeify(fs.outputFile);
...
+							errorMessage = errorMessage +
+								`\n${path.join(cwd, file)}:${line}:${column}\n`;
+						}
+						// Forward Sass error.
+						throw new Error(errorMessage);
+					}
+
+					return result.stdout;
+				}).then(css => {
+					// post css does not parse the charset unless it is also base64.
```
Comment (JakeChampion) [inline lib/tasks/build-sass.js]:
```suggestion
					// postcss does not parse the charset unless it is also base64.
```

File: lib/tasks/build.js [hunk 5]
```diff
@@ -35,22 +34,7 @@ module.exports = function (cfg) {
...
-							output += '\n' + warning.getValue().replace(/\n/g, ' ');
-						} else {
-							output = warning.getValue().replace(/\n/g, ' ');
-						}
-						task.output = output;
-						return nodeSass.NULL;
-					}
-				};
-			}
+		task: () => {
```
Comment (JakeChampion) [inline lib/tasks/build.js]: Nice! :-)

File: lib/tasks/build-sass.js [hunk 6]
```diff
@@ -31,6 +34,24 @@ function getSassData(sassFile, config = {
...
+ * @param {Boolean} config.sourcemaps [true] - Whether to include inline sourcemaps (defaults to `false` when `production` is `true`).
+ * @param {Boolean} config.production [false] - Whether this is a production build. Non-production builds do not minify CSS and include sourcemaps by default.
+ * @param {String|Undefined} config.brand [undefined] - The brand the Sass build is for .e.g. "master", "internal", or "whitelabel".
+ * @param {String|Undefined} config.sassPrefix [undefined] - Sass to prefix the Sass from file with.
+ * @param {String} config.outputStyle ['expanded'] - The Sass output style. One of `compressed` (removes as many extra characters as possible) or `expanded` (writes each selector and declaration on its own line).
+ * @param {Boolean} config.verbose [false] - When true, Sass warnings and debug messages are output.
+ * @param {String[]} config.sassIncludePaths - Extra Sass paths to includes.
+ * @param {*} config.ignoreBower - True to include npm Sass paths, otherwise bower paths are used.
+ * @return {String} - The built css.
+ */
```
Comment (JakeChampion) [inline lib/tasks/build-sass.js]: This JSDoc is too good @notlee

Comment (notlee) [reply lib/tasks/build-sass.js]: Thanks! This method could use more work, but I wanted to documented what was there to make that easier in the future

**Comment by notlee** [comment]
Best of three on my local machine: o-teaser demos build ~6% faster 😊

**Comment by notlee** [comment]
:D

**Pull request merged**